osu.Server.Queues.ScoreStatisticsProcessor/ScoreStatisticsQueueProcessor.cs

```diff
@@ -236,17 +236,11 @@ protected override void ProcessResult(ScoreItem item)
                         transaction.Commit();
                     }
 
-                    // TODO: this can be removed after https://github.com/ppy/osu-web/issues/10942 is closed out.
-                    // Intentionally not part of the transaction to avoid deadlocks.
-                    // See https://discord.com/channels/90072389919997952/983550677794050108/1199725169573380136
-                    if (score.passed)
-                    {
-                        // For now, just assume all passing scores are to be preserved.
-                        conn.Execute("UPDATE scores SET preserve = 1 WHERE id = @Id", new { Id = score.id });
-                    }
-
                     foreach (var p in enumerateValidProcessors(score))
                         p.ApplyGlobal(score, conn);
+
+                    if (!score.preserve)
+                        Console.WriteLine($"Score {score.id} was processed but not preserved");
```

Comment:
Only passed scores are initially marked as preserved, and we do process failed scores, although only partially. So this is going to start spamming immediately after deploy for every failed score, is it not? If the goal of this was to be a bulwark against expectations not being met, then it kinda doesn't work...

I'd expect something closer to

```suggestion
                    if (score.passed && !score.preserve)
                        Console.WriteLine($"Passed score {score.id} was processed but not preserved");
```